src/main/java/org/dcsa/bkg/service/impl/BKGServiceImpl.java

```diff
@@ -116,12 +116,18 @@ public Mono<BookingResponseTO> createBooking(final BookingTO bookingRequest) {
                 // collisions
                 bookingRepository.findById(bookingRefreshed.getId()))
         .flatMap(booking -> createDeepObjectsForBooking(bookingRequest, booking))
-        .flatMap(bTO -> createShipmentEventFromBookingTO(bTO).thenReturn(bTO))
+        .flatMap(
+            bTO ->
+                createShipmentEventFromBookingTO(bTO.getT1(), bTO.getT2()).thenReturn(bTO.getT2()))
         .flatMap(bTO -> Mono.just(bookingMapper.dtoToBookingResponseTO(bTO)));
   }
 
-  private Mono<BookingTO> createDeepObjectsForBooking(BookingTO bookingRequest, Booking booking) {
+  private Mono<Tuple2<UUID, BookingTO>> createDeepObjectsForBooking(
```

Comment:
Returning the tuple would not be preferable. I believe this can be solved differently.
this method can remain unchanged. 
For creating the deepobjects and the shipment events something like this can be used:
`.flatMap(
            booking ->
                createDeepObjectsForBooking(bookingRequest, booking)
                    .flatMap(
                        bookingTO ->
                            createShipmentEventFromBookingTO(booking.getId(), bookingTO)
                                .thenReturn(bookingTO)))`

Reply:
Good point. Updated to reflect suggestion.